projects/plugins/social/src/js/store/actions/jetpack-settings.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+/*eslint lodash/import-scope: [2, "method"]*/
+import pick from 'lodash/pick';
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import {
+	fetchJetpackSettings,
+	updateJetpackSettings as updateJetpackSettingsControl,
+} from '../controls';
+import { STORE_ID } from '../../store';
+
+export const SET_JETPACK_SETTINGS = 'SET_JETPACK_SETTINGS';
+export const TOGGLE_PUBLICIZE_MODULE = 'TOGGLE_PUBLICIZE_MODULE';
+
+/**
+ * Yield actions to update Publicize Settings
+ *
+ * @param {object} settings - settings to apply.
+ * @yields {object} - an action object.
+ * @returns {object} - an action object.
+ */
+export function* updateJetpackSettings( settings ) {
+	try {
+		yield setUpdatingJetpackSettings();
+		yield setJetpackSettings( settings );
+		yield updateJetpackSettingsControl( settings );
+		const updatedSettings = yield fetchJetpackSettings();
+		yield setJetpackSettings( updatedSettings );
+		return true;
+	} catch ( e ) {
+		const oldSettings = pick( select( STORE_ID ).getPublicizeModuleStatus(), [
```

Comment:
Do we need to use `pick` here? Generally I think we're trying to remove dependencies on lodash. The settings object will only be the social ones, and you can probably do something like:

```
const oldSettings = select( STORE_ID ).getPublicizeModuleStatus();
yield setJetpackSettings( { publicize_active: oldSettings?.publicize_active } );
```

Reply:
Fixed in [7b65aa4](https://github.com/Automattic/jetpack/pull/24203/commits/7b65aa4d1e005e21a28fa70435db978b3711a9a8)